### DISCUSSION TITLE
Chat Space

### DISCUSSION THREAD

**Discussion by Avdhesh-Varshney** [code-a2z/code-a2z#139] [General]
Hello Open-source Contributors,
This is a general communication channel for all the open-source contributors of this project. 
If you have any kind of query, feel free to drop the message in this forum.

**Comment by Zunairah-k** [comment]
@Avdhesh-Varshney I’m a new contributor for Code A2Z under GSSoC’25. I’ve added the Sign Out confirmation modal in UserNavigationPanel.jsx, but I’m unable to run the frontend locally because the app crashes (blank page) due to API/backend dependencies. Could you guide me on the best way to test my changes locally before submitting a PR?

**Reply by Avdhesh-Varshney** [reply]
Yes you can use .env with BE deploy URL
https://code-a2z.onrender.com

**Reply by Zunairah-k** [reply]
The .env file is already set to the deployed backend, but the frontend still shows a blank page locally

**Reply by Avdhesh-Varshney** [reply]
Share any ss

**Reply by Zunairah-k** [reply]
![Screenshot_20250909_202550.jpg](https://github.com/user-attachments/assets/78a7175d-7e6e-4632-be31-4645650797fa)

![Screenshot_20250909_202539.jpg](https://github.com/user-attachments/assets/21a31df4-27af-4cce-bfc1-7c711f92dc30)


here's what I see when I run the project locally: the page is blank in Chrome, and this is what the terminal shows for the server running on localhost. Not sure why the frontend isn't loading the data.

**Reply by Zunairah-k** [reply]
Also please check the back to top redirection in read me PR and merge when possible

**Reply by Avdhesh-Varshney** [reply]
https://github.com/Code-A2Z/code-a2z/blob/main/SETUP.md

Check this ☝️ may help

**Reply by Zunairah-k** [reply]
Okay, I'll try the steps again

**Comment by Avdhesh-Varshney** [comment]
@100NikhilBro We can discuss here in this thread!

**Reply by Avdhesh-Varshney** [reply]
Right now! I'm looking to transform the whole project from js to ts and also organize the code into modular structure for whole FE & BE

**Reply by 100NikhilBro** [reply]
Hi @Avdhesh-Varshney ,
 I’m ready for it bro 🚀 I’ll work on transforming the whole project from JS to TS and also focus on organizing the code into a proper modular structure across both frontend and backend. I’ll make sure it’s cleaner, scalable, and easier to maintain going forward 💯

**Reply by Avdhesh-Varshney** [reply]
Thanks bro! I've done with many changes and structuring, check #1206 
This is the PR from where you need to start so plz clone that branch and review the code structure.

`client/src/App.tsx` file 👇 

```
<Route path="/" element={<Navbar />}>
     <Route index element={<Home />} />
     <Route path="login" element={<UserAuthForm type="login" />} />
     <Route path="signup" element={<UserAuthForm type="signup" />} />
     <Route path="search/:query" element={<Search />} />
     <Route path="user/:id" element={<Profile />} />
     <Route path="project/:project_id" element={<Project />} />

     <Route path="dashboard" element={<Sidebar />}>
     </Route>
     <Route path="settings" element={<Sidebar />}>
     </Route>

</Route>

<Route path="/editor" element={<Editor />} />
<Route path="/editor/:project_id" element={<Editor />} />

<Route path="*" element={<PageNotFound />} />
```

Left parts are routes under sidebar, which you can go through `frontend/src/App.jsx` file

**Reply by 100NikhilBro** [reply]
Okk Bro , I check it

**Reply by Avdhesh-Varshney** [reply]
@100NikhilBro Any blocker or any question?

**Reply by 100NikhilBro** [reply]
Hi @Avdhesh-Varshney,
Got it bhai 🙌 — I’ll check it as soon as possible. Meanwhile, you can continue the work.

Thanks for pointing out #1206, I’ll start the review from there and focus on the updated code structure.
Noted about the client/src/App.tsx routes too 👍

**Reply by Avdhesh-Varshney** [reply]
@100NikhilBro I've transform the whole BE to TS. But still deployment is failed due to not able to install dependency related to TS files. Could you try something from my ts server branch?

**Reply by 100NikhilBro** [reply]
@Avdhesh-Varshney Bro, I didn’t touch anything on that branch.

**Reply by Avdhesh-Varshney** [reply]
No I'm telling you. If you have time, could you check the deployment issue, server will be merge

**Reply by 100NikhilBro** [reply]
Bro, if I find some time, I’ll definitely look into the deployment issue.

**Reply by Avdhesh-Varshney** [reply]
@100NikhilBro You can resolve your assigned issues now.
I've just moving with JS right now. As there are a lots of deployment issues in TS server.

Happy Contribution 🎉

**Reply by 100NikhilBro** [reply]
@Avdhesh-Varshney I’ll resolve all the issues within this week. After that, once I’m free, we can discuss and try migrating the whole backend to TypeScript if possible.

**Reply by Avdhesh-Varshney** [reply]
Great
